src/cli.js

```diff
@@ -49,11 +48,11 @@ module.exports = async () => {
   } catch (err) {
     // If error is a SemanticReleaseError then it's an expected exception case (no release to be done, running on a PR etc..) and the cli will return with 0
     // Otherwise it's an unexpected error (configuration issue, code issue, plugin issue etc...) and the cli will return 1
-    if (err instanceof SemanticReleaseError) {
+    if (err.name === 'SemanticReleaseError') {
```

Comment:
If plugins want to extend the SemanticReleaseError the name will change, see @felixfbecker's question at https://github.com/semantic-release/error/pull/71#discussion_r147571577?

What was your thinking behind changing the condition to check for the name?

Reply:
Yes that's a problem indeed. Maybe the safe side solution is `if (err.name === 'SemanticReleaseError' || err instanceof SemanticReleaseError) {`.

@felixfbecker what do you think?

Reply:
Thinking about it best solution might be to set a property in the SemanticReleaseError that indicate it's a semantic-release error. This way it would work with inheritance and wouldn't have the issue of multiple version of the error being different instances (and failing the `err instanceof SemanticReleaseError` test).

Reply:
What we need here is a common interface to be implemented by all errors that should not cause the build to not fail / are "expected". This could be
- common `name` that is always `SemanticReleaseError`, even for subclasses
- a property like `expected: true`
- a property `exitCode: 0` (this is non ideal because ChildProcess errors have an `exitCode` property)

Reply:
Yes. See https://github.com/semantic-release/error/pull/72

Once that is merge I can do `if (err.semanticRelease) {`.

I choose to name the property `semanticRelease` instead of `expected` or `exitCode` to avoid conflicts.

Reply:
love it, good solution 👍

Reply:
Ok. Once we merge semantic-release/error#72 it would be a good idea to deprecate older versions, to make sure third-party plugins use the last version that include the property.

Reply:
Modification done